### PR TITLE
Cosmos client mnemonic

### PR DESCRIPTION
**Pull request by expertdicer**
This Pull request includes:

- Change client key input, you now only need to input ``mnemonic`` instead of all 4 items ``private key``, ``public key``, ``account``, ``address``.
- Test cases for the ``mnemonic`` feature.

### CODE REVIEW

**Comment by expertdicer** [comment]
@blasrodri  @vmarkushin Can you guys look into this ?

File: hyperspace/cosmos/src/client.rs [hunk 1]
```diff
@@ -70,6 +81,28 @@ impl TryFrom<ConfigKeyEntry> for KeyEntry {
...
 }
 
+impl TryFrom<String> for KeyEntry {
+	type Error = bip32::Error;
+
+	fn try_from(mnemonic: String) -> Result<Self, Self::Error> {
+		// From mnemonic to pubkey
+		let mnemonic = bip39::Mnemonic::from_phrase(&mnemonic, bip39::Language::English).unwrap();
+		let seed = bip39::Seed::new(&mnemonic, "");
+		let key_m = XPrv::derive_from_path(seed, &DerivationPath::from_str("m/44'/118'/0'/0/0")?)?;
```
Comment (blasrodri) [inline hyperspace/cosmos/src/client.rs:91]: Can you provide a reference that explains where this path comes from!

**Review by blasrodri — APPROVED** [review]

**Comment by blasrodri** [comment]
You need to run cargo +nightly fmt I believe

**Review by blasrodri — APPROVED** [review]

File: hyperspace/config.toml [hunk 2]
```diff
@@ -1,10 +1,10 @@
-# Parachain to Parachain configuration template
+# Parachain to Cosmos configuration template
 [chain_a]
 type = "parachain"
 name = "picasso"
 para_id = 2001
-parachain_rpc_url = "ws://127.0.0.1:9988"
-relay_chain_rpc_url = "ws://127.0.0.1:9944"
+parachain_rpc_url = "ws://65.108.73.88:9988"
```
Comment (blasrodri) [inline hyperspace/config.toml:6]: we need to revert this change

File: hyperspace/cosmos/Cargo.toml [hunk 3]
```diff
@@ -29,6 +29,12 @@ tonic = { version = "0.8", features = ["tls", "tls-roots"] }
 bech32 = "0.9.1"
 bip32 = "0.4.0"
 ed25519-zebra = { version = "3.1.0" }
+tiny-bip39 = "1.0.0"
```
Comment (blasrodri) [inline hyperspace/cosmos/Cargo.toml:32]: are you sure we need all these dependencies?

File: hyperspace/config.toml (hunk 2 above)
Comment (expertdicer) [reply hyperspace/config.toml:6]: Yes. I agree

File: hyperspace/cosmos/Cargo.toml (hunk 3 above)
Comment (expertdicer) [reply hyperspace/cosmos/Cargo.toml:32]: The ``data-encoding`` and ``secp256k1 `` one is redundant. Sorry for the mistake

**Review by vmarkushin — APPROVED** [review]

**Comment by blasrodri** [comment]
@expertdicer can you squash all commits into one and sign it?

**Comment by expertdicer** [comment]
> @expertdicer can you squash all commits into one and sign it?

Does it look ok now ?

**Review by blasrodri — APPROVED** [review]

**Comment by blasrodri** [comment]
It needs to have all commits signed. The first isn't. That is why my suggestion is that you take the three commits, squash them into one, and sign that new one.

**Comment by expertdicer** [comment]
Hey @blasrodri . It involves many merge conflict and I can't solve it. So I figure I will create a new branch and a new pr to save time.
